### PR TITLE
Move libs from ouroboros-network

### PR DESCRIPTION
**Pull request by nc6**
These modules come from `ouroboros-network` and are being moved here to support moving some other things to `cardano-base`.

### CODE REVIEW

File: src/Data/FingerTree/Strict.hs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+-- | Strict variants of 'FingerTree' operations.
+--
+-- TODO: Move this to @cardano-prelude@ as a new module, @Data.FingerTree.Strict@.
```
Comment (mrBliss) [inline src/Data/FingerTree/Strict.hs]: This TODO can go now :slightly_smiling_face:

File: src/Data/Sequence/Strict.hs [hunk 2]
```diff
@@ -0,0 +1,224 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE PatternSynonyms            #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE ViewPatterns               #-}
+
+-- | Strict variants of 'Seq' operations.
+--
+-- TODO: Move this to @cardano-prelude@ as a new module, @Data.Sequence.Strict@.
```
Comment (mrBliss) [inline src/Data/Sequence/Strict.hs]: This one too

File: src/Data/FingerTree/Strict.hs (hunk 1 above)
Comment (nc6) [reply src/Data/FingerTree/Strict.hs]:
```suggestion
```

File: src/Data/Sequence/Strict.hs (hunk 2 above)
Comment (nc6) [reply src/Data/Sequence/Strict.hs]:
```suggestion
```

**Pull request merged**